src/humanloop/eval_utils.py

```diff
@@ -0,0 +1,543 @@
+            for datapoint in hl_dataset.datapoints
+        ]
+        for _ in as_completed(futures):
+            completed_tasks += 1
+            _progress_bar(total_datapoints, completed_tasks)
+
+    # Wait for the Evaluation to complete then print the results
+    complete = False
+    stats = None
+    while not complete:
+        stats = client.evaluations.get_stats(id=evaluation.id)
+        sys.stdout.write(stats.progress)
+        sys.stdout.flush()
+        complete = stats.status == "completed"
+        if not complete:
+            time.sleep(5)
+
+    # Print Evaluation results
+    print(stats.report)
+
+    checks: list[EvaluatorCheck] = []
+    for evaluator in evaluators:
+        improvement_check, score, delta = check_evaluation_improvement(
+            evaluation=evaluation,
+            stats=stats,
+            evaluator_path=evaluator["path"],
+        )
+        threshold_check = None
+        threshold = evaluator.get("threshold")
+        if threshold is not None:
+            threshold_check = check_evaluation_threshold(
+                evaluation=evaluation,
+                stats=stats,
+                evaluator_path=evaluator["path"],
+                threshold=threshold,
+            )
+        checks.append(
+            EvaluatorCheck(
+                path=evaluator["path"],
+                improvement_check=improvement_check,
+                score=score,
+                delta=delta,
+                threshold=threshold,
+                threshold_check=threshold_check,
+            )
+        )
+    return checks
+
+
+def _get_log_func(
+        client: BaseHumanloop,
+        type_: Literal["flow", "prompt", "tool", "evaluator"],
+        file_id: str,
+        version_id: str,
+        evaluation_id: str,
+        batch_id: str,
+) -> Callable:
+    """Returns the appropriate log function pre-filled with common parameters."""
+    log_request = {
+        # TODO: why does the Log `id` field refer to the file ID in the API?
+        #  Why are both `id` and `version_id` needed in the API?
+        "id": file_id,
+        "version_id": version_id,
+        "evaluation_id": evaluation_id,
+        "batch_id": batch_id,
+    }
+    match type_:
+        case "flow":
+            return partial(client.flows.log, **log_request, trace_status="complete")
+        case "prompt":
+            return partial(client.prompts.log, **log_request)
+        case "evaluator":
+            return partial(client.evaluators.log, **log_request)
+        case "tool":
+            return partial(client.tools.log, **log_request)
+        case _:
+            raise NotImplementedError(f"Unsupported File version: {type_}")
+
+
+def get_score_from_evaluator_stat(stat: NumericStats | BooleanStats) -> float | None:
+    """Get the score from an Evaluator Stat."""
+    score = None
+    match stat:
+        case BooleanStats():
+            if stat.total_logs:
+                score = round(stat.num_true / stat.total_logs, 2)
```

Comment:
Is it possible for a dataset to have no logs and add it to the evaluation? Wondering if this could result in a divide by zero